### PR TITLE
Extract the GUI from IML

### PR DESCRIPTION
**Pull request by jgrund**
Extract the IML GUI and fetch it as a RPM instead.

### CODE REVIEW

File: chroma-manager/chroma-manager.conf.template [hunk 1]
```diff
@@ -32,7 +32,7 @@ server {
     gzip_comp_level 2;
 
     location /favicon.ico {
-      alias {{APP_PATH}}/ui-modules/node_modules/@iml/gui/dist/source/intel.ico;
+      alias /usr/lib/iml-manager/gui/dist/source/intel.ico;
```
Comment (johnsonw) [inline chroma-manager/chroma-manager.conf.template]: I thought it was iml-manager/iml-gui?

**Comment by brianjmurrell** [comment]
Maybe just still to come, but will this patch include excising the GUI build stuff from the IML build process?

**Comment by jgrund** [comment]
> Maybe just still to come, but will this patch include excising the GUI build stuff from the IML build process?

Not yet, because there are still a few more node modules being built into the resultant manager RPM.

Everything under `dependencies` [here](https://github.com/intel-hpdd/intel-manager-for-lustre/blob/03c398d1a27b1d9f1b6f859b08b5d79cb115f787/chroma-manager/ui-modules/package.json) needs to be repackaged.

On the plus side, extracting this module drops us to 5.35MB for the tar file. The vast majority of the bulk is in ui-modules/node_modules, so once that's gone the tar file will be tiny.

screen shot 2017-07-25 at 9 38 06 am



A few more can be extracted without more work, but the rest are going to need a configuration service to get some values. [Consul](https://www.consul.io/) looks interesting for this.

File: chroma-manager/tests/framework/utils/install_deps.sh [hunk 2]
```diff
@@ -4,4 +4,5 @@ mkdir -p chroma_test_env
 tar -C chroma_test_env -xizf -"
 
 ssh root@$CHROMA_MANAGER "set -ex
-yum -y install nodejs npm nginx libuv"
+yum-config-manager --add-repo https://copr.fedorainfracloud.org/coprs/managerforlustre/manager-for-lustre/repo/epel-7/managerforlustre-manager-for-lustre-epel-7.repo
+yum -y install nodejs npm nginx libuv iml-gui iml-srcmap-reverse"
```
Comment (brianjmurrell) [inline chroma-manager/tests/framework/utils/install_deps.sh]: _iml-gui_ and _iml-srcmap-reverse_ are supposed to get installed when the manager is installed through the _Requires:_ additions to the _chroma-manager.spec_ changes above.

But that should happen in production environments also.  Adding the Copr repo for those packages in this file only adds it in the test environment.

What you actually [need](https://github.com/intel-hpdd/intel-manager-for-lustre/pull/142/files#diff-ac0571f67c3363c9e3acdc025d81fe71R366) is in another [WiP patch](#142).

Comment (jgrund) [inline chroma-manager/tests/framework/utils/install_deps.sh]: This is only used in "development" setups for unit and service tests. There is no installation of  iml in a production way done there.

Comment (brianjmurrell) [inline chroma-manager/tests/framework/utils/install_deps.sh]: Ahhh.  right.  Thanks for jogging my memory on that.

I think my concerns about the MFL Copr repo during manager install still stand though.  I'll restate them in this patch where it's relevant.

**Pull request merged**